### PR TITLE
PYTHON-2965 Migrate to a PEP517 compliant build system

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -4,10 +4,6 @@
 import sys
 import warnings
 
-if sys.version_info[:3] < (3, 7):
-    raise RuntimeError("Python version >= 3.7 required.")
```
Comment (ShaneHarvey) [inline setup.py:8]: Do we still need this as long as setup.py exists?

File: setup.py [hunk 2]
```diff
@@ -25,29 +21,6 @@
...
-f = open("README.rst")
-try:
-    try:
-        readme_content = f.read()
-    except BaseException:
-        readme_content = ""
-finally:
-    f.close()
-
 # PYTHON-654 - Clang doesn't support -mno-fused-madd but the pythons Apple
```
Comment (ShaneHarvey) [inline setup.py:24]: How does the new build process handle this "-mno-fused-madd" issue? Or is it no longer a concern.

File: setup.py (hunk 1 above)
Comment (blink1073) [reply setup.py:8]: No, setuptools will use the `python_requires` config from `pyproject.toml`.

File: setup.py (hunk 2 above)
Comment (blink1073) [reply setup.py:24]: Actually, we don't need to use the stdlib `distutils` at all, and the `distutils` that ships with `setuptools` does not add this flag.  I cleaned up this block.

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**